src/render/filters.rs

```diff
@@ -308,6 +310,89 @@ impl ResolveFilter for ExternalFilter {
     }
 }
 
+impl ResolveFilter for FirstFilter {
+    fn resolve<'t, 'py>(
+        &self,
+        variable: Option<Content<'t, 'py>>,
+        py: Python<'py>,
+        _template: TemplateString<'t>,
+        _context: &mut Context,
+    ) -> ResolveResult<'t, 'py> {
+        let content = match variable {
+            Some(content) => content,
+            None => return Ok(Some("".as_content())),
+        };
+
+        match content {
+            Content::Py(obj) => {
+                // Try to get the first item using Python's indexing
+                // Django only catches IndexError, not TypeError
+                match obj.get_item(0) {
+                    Ok(first) => Ok(Some(Content::Py(first))),
+                    Err(e) if e.is_instance_of::<PyIndexError>(py) => Ok(Some("".as_content())),
+                    Err(e) if e.is_instance_of::<pyo3::exceptions::PyTypeError>(py) => {
+                        // Check if this is the standard "'type' object is not subscriptable" error
+                        let error_msg = e.to_string();
+                        if error_msg.contains("is not subscriptable") {
+                            // Standard not subscriptable error - provide better formatting
+                            let type_name = obj
+                                .get_type()
+                                .name()
+                                .map(|n| n.to_string())
+                                .unwrap_or_else(|_| "unknown".to_string());
+                            Err(RenderError::NotSubscriptable {
+                                type_name,
+                                at: self.at,
+                            }
+                            .into())
+                        } else {
+                            // Custom TypeError - preserve the original message
+                            Err(e.into())
+                        }
+                    }
+                    Err(e) => Err(e.into()),
```

Comment:
Since my last review, I've added a [`PyErr::annotate` method](https://github.com/LilyAcorn/django-rusty-templates/blob/49b96d65a672d2ec09b48885a83b89f2bea841ef/src/error.rs#L84-L98) which I think we can use here:

```suggestion
                    Err(error) => {
                        let error = error.annotate(py, self.at, "here", template);
                        Err(error.into())
                    }
```